alex/applications/PublicTransportInfoCS/hdc_policy.py

```diff
@@ -572,7 +572,37 @@ def get_requested_info(self, requested_slots, ds, accepted_slots):
         res_da = DialogueAct()
 
         for slot in requested_slots:
-            if ds['route_alternative'] in [0, 1, 2, 3]:
+            # for these, we don't need any route information, we can answer straight away
+            if slot not in ['from_stop', 'to_stop',
+                            'departure_time', 'departure_time_rel',
+                            'arrival_time', 'arrival_time_rel',
+                            'duration', 'num_transfers', 'time_transfers', ]:
+                dai = DialogueActItem("inform", slot, requested_slots[slot])
+                res_da.append(dai)
+                # remember that we answered
+                ds["rh_" + slot].reset()
+                continue
+
+            # try to find a route if we don't know it yet
+            if not isinstance(ds['route_alternative'], int):
+                req_da, iconfirm_da, conn_info = self.gather_connection_info(ds, accepted_slots)
+                # we have all information we need, start searching
+                if len(req_da) == 0:
+                    ds.conn_info = conn_info
+                    res_da = iconfirm_da
+                    # the search will change ds['route_alternative'] if a route is found
+                    dir_da = self.get_directions(ds, check_conflict=True)
+                    # only return the output DAs if no route is found (i.e., the error message),
+                    # otherwise we go on to return the specific information requested
+                    if not isinstance(ds['route_alternative'], int):
+                        res_da.extend(dir_da)
+                # we don't know enough, ask about the rest
+                else:
+                    res_da = req_da
+
+            # we have a route, so return information about it
+            # NB: ds['route_alternative'] might have changed in the meantime if a route has been found
```

Comment:
I added this note and a fix on L595-598 that might make it more explicit that the `if`s aren't mutually exclusive.